cli.go

```diff
@@ -198,6 +191,48 @@ func (i *CLI) Run() {
 	}
 }
 
+func (i *CLI) handleStartHook() {
+	hookValue := i.c.String("start-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("start_hook", hookValue).Info("running")
+
+	parts := stringSplitSpace(hookValue)
+	outErr, err := exec.Command(parts[0], parts[1:]...).CombinedOutput()
+	if err == nil {
+		return
+	}
+
+	i.logger.WithFields(logrus.Fields{
+		"err":        err,
+		"output":     string(outErr),
+		"start_hook": hookValue,
+	}).Error("start hook failed")
+}
+
+func (i *CLI) handleStopHook() {
+	hookValue := i.c.String("stop-hook")
+	if hookValue == "" {
+		return
+	}
+
+	i.logger.WithField("stop_hook", hookValue).Info("adding deferred execution")
```

Comment:
This line doesn't seem correct anymore. It used to be called before we did the `defer`, but now it's run "inside" of the `defer`-ed function.